mountpoint-s3/src/fs.rs

```diff
@@ -49,37 +47,45 @@ impl DirHandle {
 }
 
 #[derive(Debug)]
-struct FileHandle<Client: ObjectClient, Runtime> {
```

Comment:
Is making this a type parameter going to be annoying when it comes to switching caching on/off? I was imagining that maybe we're choose a different `ObjectStore` implementation based on whether the cache is enabled or not, but maybe that's not what you had in mind.

edit: oh, I see now, you statically switch on it with `create_filesystem`. That's probably fine, although note that we'll end up with two copies of a bunch of code in the final binary (everything with a `Store` generic).